blocks/subscriptions-block/components/useOffer.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+const offerService = ({
+  origin,
+  code,
+  endpoint,
+}) => fetch(`${origin}${endpoint}${code || 'default'}`, {})
+  .then((res) => res.json());
+
+const useOffer = ({ campaignCode }) => {
+  const { arcSite } = useFusionContext();
+  const { api: { retail: { origin = null, endpoint } } } = getProperties(arcSite);
+  const [offer, setOffer] = useState(null);
+  const [error, setError] = useState(null);
+  const [isFetching, setIsFetching] = useState(true);
+
+  const fetchOffer = useCallback(async (code) => {
+    try {
```

Comment:
possibly check here if `origin` and `endpoint` are truthy before following through.  If not, `setError()` ?